src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesLauncher.java

```diff
@@ -293,6 +312,7 @@ private Pod getPodTemplate(KubernetesSlave slave, PodTemplate template) {
                 .withContainers(containers.values().toArray(new Container[containers.size()]))
                 .withNodeSelector(getNodeSelectorMap(template.getNodeSelector()))
                 .withRestartPolicy("Never")
+                .withAffinity(kubernetesPodAffinity)
```

Comment:
maybe this can't be added if affinity is not set ?

Reply:
I went over the code, and initially, if there is no `node affinity`, `pod affinity` or `pod antiaffinity`, we give an `Affinity` without setting anything. This is how the underying object behaviour of the `Affinity` object in the kubernetes fabric8 model.